io/include/pcl/compression/color_coding.h

```diff
@@ -45,357 +45,353 @@
+        diffBlue = static_cast<unsigned char> (diffBlue << colorBitReduction_);
+
+        // decode color information
+        colorInt = ((avgRed ^ diffRed) << 0) |
+                   ((avgGreen ^ diffGreen) << 8) |
+                   ((avgBlue ^ diffBlue) << 16);
+      }
+      else
+      {
+        // decode color information
+        colorInt = (avgRed << 0) | (avgGreen << 8) | (avgBlue << 16);
+      }
+
+      char* idxPointPtr = reinterpret_cast<char*> (&outputCloud_arg->points[beginIdx_arg + i]);
+      int& pointColor = *reinterpret_cast<int*> (idxPointPtr+rgba_offset_arg);
+      // assign color to point from point cloud
+      pointColor=colorInt;
+    }
+  }
+
+  /** \brief Set default color to points
+   * \param outputCloud_arg output point cloud
+   * \param beginIdx_arg index indicating first point to be assigned with color information
+   * \param endIdx_arg index indicating last point to be assigned with color information
+   * \param rgba_offset_arg offset to color information
+   * */
+  void
+  setDefaultColor (PointCloudPtr outputCloud_arg, std::size_t beginIdx_arg, std::size_t endIdx_arg, unsigned char rgba_offset_arg)
+  {
+    assert (beginIdx_arg <= endIdx_arg);
+
+    // amount of points to be decoded
+    unsigned int pointCount = static_cast<unsigned int> (endIdx_arg - beginIdx_arg);
+
+    // iterate over points
+    for (std::size_t i = 0; i < pointCount; i++)
+    {
+      char* idxPointPtr = reinterpret_cast<char*> (&outputCloud_arg->points[beginIdx_arg + i]);
+      int& pointColor = *reinterpret_cast<int*> (idxPointPtr+rgba_offset_arg);
+      // assign color to point from point cloud
+      pointColor = defaultColor_;
+    }
   }
-}
 
-#define PCL_INSTANTIATE_ColorCoding(T) template class PCL_EXPORTS pcl::octree::ColorCoding<T>;
+protected:
+  /** \brief Pointer to output point cloud dataset. */
+  PointCloudPtr output_;
+
+  /** \brief Vector for storing average color information  */
+  std::vector<char> pointAvgColorDataVector_;
+
+  /** \brief Iterator on average color information vector */
+  std::vector<char>::const_iterator pointAvgColorDataVector_Iterator_;
+
+  /** \brief Vector for storing differential color information  */
+  std::vector<char> pointDiffColorDataVector_;
+
+  /** \brief Iterator on differential color information vector */
+  std::vector<char>::const_iterator pointDiffColorDataVector_Iterator_;
+
+  /** \brief Amount of bits to be removed from color components before encoding */
+  unsigned char colorBitReduction_;
+
+  // frame header identifier
+  static const int defaultColor_;
+};
+
+// define default color
+template<typename PointT>
+const int ColorCoding<PointT>::defaultColor_ = ((255) << 0) |
+                                               ((255) << 8) |
+                                               ((255) << 16);
+
+#define PCL_INSTANTIATE_ColorCoding(T) template class PCL_EXPORTS ColorCoding<T>;
```

Comment:
Maybe move this out for uniformity? So all instantiations not depending on Point types don't throw error.